common/flatpak-utils.c

```diff
@@ -3122,6 +3122,46 @@ flatpak_complete_word (FlatpakCompletion *completion,
   g_print ("%s\n", rest);
 }
 
+static gboolean
+switch_already_in_line (FlatpakCompletion *completion,
+                        GOptionEntry      *entry)
+{
+  guint i = 0;
+  guint line_part_len = 0;
+
+  for (; i < completion->original_argc; ++i)
+    {
+      line_part_len = strlen (completion->original_argv[i]);
+      if (line_part_len > 2 &&
+          g_strcmp0 (&completion->original_argv[i][2], entry->long_name) == 0)
```

Comment:
I'm not sure why you look at strlen only. You need to verify that it starts with "--", not just anything that is two chars.
I would recommend replacing the whole strlen check with g_str_has_prefix (completion->original_argv[i], "--"), etc.